lib/git/config.go

```diff
@@ -35,10 +42,12 @@ func CompileAncestorBranches(branchName string) (result []string) {
 // DeleteAllAncestorBranches removes all Git Town ancestor entries
 // for all branches from the configuration.
 func DeleteAllAncestorBranches() {
-	configs := util.GetCommandOutput("git", "config", "--get-regexp", "^git-town-branch\\..*\\.ancestors$")
-	for _, config := range strings.Split(configs, "\n") {
-		splitConfig := strings.Split(config, " ")
-		removeConfigurationValue(splitConfig[0])
+	configRegexp, err := regexp.Compile("^git-town-branch\\..*\\.ancestors$")
+	if err != nil {
+		log.Fatal("Error compiling ancestor branches configuration regular expression: ", err)
+	}
+	for _, key := range getConfigurationKeysMatching(configRegexp) {
```

Comment:
What about making `getConfigurationKeysMatching` take in a string and compile the regex in there? The code to compile regex is duplicated in a number of places.